src/fusion/trash.nim

```
@@ -0,0 +1,113 @@
+## Convenience procedures for the operating system Trash support.
+## Allows to trash several files with the same name.
+##
+## * http://www.freedesktop.org/wiki/Specifications/trash-spec
+## * http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html
+when defined(js):
+  {.fatal: "Module trash is not designed to be used with the JavaScript backend.".}
+
+import os, times, strutils
+
+const trashDirDefault {.strdefine, used.}: string = ""
+const trashinfo = """
+[Trash Info]
+Path=$1
+DeletionDate=$2
+"""
+
+
+template trashHelper(trashPath: string; fname: string): string =
+  when defined(linux) or defined(bsd): trashPath / "files" / fname
+  else: trashPath / fname
+
+
+proc getTrash*(): string =
+  ## Return the operating system Trash directory.
+  ## Android has no Trash, some apps just use a temporary folder.
+  ## Windows Trash is not supported.
+  ## Use `-d:trashDirDefault=/path/to/trash` for not supported operating systems.
+  result =
+    when defined(linux) or defined(bsd):
+      getEnv("XDG_DATA_HOME", getHomeDir()) / ".local/share/Trash"
+    elif defined(osx):
+      getEnv("HOME", getHomeDir()) / ".Trash"
+    elif defined(android):
+      getTempDir()
+    else:
+      when trashDirDefault.len > 0:
+        trashDirDefault
+      else:
+        doAssert false, "Operating system Trash is currently not supported, use -d:trashDirDefault=path"
+
```

validate `result` here (in particular to guard against wrong `trashDirDefault` and fail early, but not just this case)
https://specifications.freedesktop.org/trash-spec/trashspec-latest.html

> The implementation also MUST check that this directory is not a symbolic link.

(and IMO should be an absolute dir)

---

`Network paths are considered absolute too.`  :thinking: 
https://nim-lang.github.io/Nim/os.html#isAbsolute%2Cstring

---

network paths seem allowed in spec (and in my experience as well):

> The implementation MAY also support trashing files from the rest of the system (including other partitions, shared network resources, and removable devices) into the “home trash” directory .

in anycase, `os.isNetworkPath` would indeed be useful refs https://github.com/timotheecour/Nim/issues/467